pandas/tests/internals/test_internals.py

```diff
@@ -1164,63 +1164,117 @@ def any(self, axis=None):
+
+        result = op(s, e.value).dtypes
+        expected = op(s, value).dtypes
+        tm.assert_series_equal(result, expected)
+
+    @ops
+    @pytest.mark.parametrize("value", [1j, 2 ** 63])
+    def test_binop_complex128(self, op, value):
+        dtype = "complex128"
+
+        if op == operator.mod:
+            pytest.skip(f"Invalid combination {op},{dtype}")
+
+        e = DummyElement(value, dtype)
+        s = pd.DataFrame({"A": [e.value, e.value]}, dtype=e.dtype)
+
+        result = op(s, e.value).dtypes
+        expected = op(s, value).dtypes
+        tm.assert_series_equal(result, expected)
+
+    @ops
+    @pytest.mark.parametrize("value", [1.0, 2 ** 63])
+    def test_binop_f8(self, op, value):
+        dtype = "f8"
 
         e = DummyElement(value, dtype)
         s = pd.DataFrame({"A": [e.value, e.value]}, dtype=e.dtype)
 
-        invalid = {
-            (operator.pow, "<M8[ns]"),
-            (operator.mod, "<M8[ns]"),
-            (operator.truediv, "<M8[ns]"),
-            (operator.mul, "<M8[ns]"),
-            (operator.add, "<M8[ns]"),
-            (operator.pow, "<m8[ns]"),
-            (operator.mul, "<m8[ns]"),
-        }
-
-        if (op, dtype) in invalid:
+        result = op(s, e.value).dtypes
+        expected = op(s, value).dtypes
+        tm.assert_series_equal(result, expected)
+
+    @ops
+    @pytest.mark.parametrize("value", [1])
+    def test_binop_i8(self, op, value):
+        dtype = "i8"
+
+        if op == operator.mod:
+            pytest.skip(f"Invalid combination {op},{dtype}")
+
+        e = DummyElement(value, dtype)
+        s = pd.DataFrame({"A": [e.value, e.value]}, dtype=e.dtype)
+
+        result = op(s, e.value).dtypes
+        expected = op(s, value).dtypes
+        tm.assert_series_equal(result, expected)
+
+    @ops
+    @pytest.mark.parametrize("value", [np.timedelta64(20, "ns")])
+    def test_binop_m(self, op, value):
+        dtype = "<m8[ns]"
+
+        e = DummyElement(value, dtype)
+        s = pd.DataFrame({"A": [e.value, e.value]}, dtype=e.dtype)
+
+        if op in [operator.mul, operator.pow]:
+            with pytest.raises(TypeError):
+                op(s, e.value)
+        else:
+            result = op(s, e.value).dtypes
+            expected = op(s, value).dtypes
+            tm.assert_series_equal(result, expected)
+
+    @ops
+    @pytest.mark.parametrize("value", [np.datetime64(20, "ns")])
+    def test_binop_m_capitalized(self, op, value):
+        dtype = "<M8[ns]"
+
+        e = DummyElement(value, dtype)
+        s = pd.DataFrame({"A": [e.value, e.value]}, dtype=e.dtype)
+
+        if op in [
+            operator.add,
+            operator.mod,
+            operator.mul,
+            operator.pow,
+            operator.truediv,
+        ]:
             with pytest.raises(TypeError):
                 op(s, e.value)
         else:
-            # FIXME: Since dispatching to Series, this test no longer
```

Comment:
Removed because I think this was correct for when this test was testing all of the operators and also multiple dtypes.

I don't know if this needs to be copied for every (new) function?

And besides if this is not testing anything useful, what will be a useful test>